snooty/parser.py

```diff
@@ -161,8 +161,13 @@ def __call__(
             options["checksum"] = checksum
             dimensions = self.asset.dimensions
             if dimensions is not None:
-                options["width"] = str(dimensions[0])
-                options["height"] = str(dimensions[1])
+                user_width = options.get("width")
+                if user_width is None:
+                    options["width"] = str(dimensions[0])
+                    options["height"] = str(dimensions[1])
+                else:
+                    width_num = float(re.sub(r"\D+", "", user_width))
```

Comment:
I don't think it's a problem in this case, but *generally* speaking you want to be careful using the top-level ad-hoc `re` functions because they're implemented with a small capped cache of compiled patterns that just clears once ~100 entries are hit, so you can end up recompiling the same pattern many times

Reply:
great point! moved to module level